### ISSUE TITLE
Error: Failed to find a python interpreter in the .data section

### ISSUE THREAD

**Issue by secsilm** [benfred/py-spy#546]
I got the `Error: Failed to find a python interpreter in the .data section` error when I run `sudo env "PATH=$PATH" RUST_LOG=debug py-spy record -o profile.svg --pid 40067`.

[Here](https://github.com/benfred/py-spy/files/10369974/log.txt) is the full log.

Here is my environment:

- OS: Ubuntu 18.04.5 LTS (Bionic Beaver)
- Python: Python 3.8.0
- py-spy: 0.3.14

**Comment by benfred** [comment]
How did you install the python you are trying to profile ?

**Comment by Atleago** [comment]
Not the reporter, and we're on Windows here:

---

```
Python 3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> import sys; print(sys.version)
3.10.5 (tags/v3.10.5:f377153, Jun  6 2022, 16:14:13) [MSC v.1929 64 bit (AMD64)]
```

This python is downloaded from [www.python.org](https://www.python.org/downloads/windows/), installed via PyCharm.

py-spy is installed through pip.
```
> py-spy.exe --version
py-spy 0.3.14
```

**Comment by secsilm** [comment]
> How did you install the python you are trying to profile ?

Sorry for late reply. I install the python and pip by:

```
sudo apt install python3.8
sudo apt install python3-pip
python3.8 -m pip install pip
```

**Comment by martin-kokos** [comment]
Same error here.
command `py-spy record -o profile.svg -- python ccdata.py `

My environment:
- OS: Gentoo
- Python: Python 3.11.1 (compiled via distro package manager, GCC 12.2.1 20230121 )
- py-spy: 0.3.14 (via poetry)


compiled with:
`x86_64-pc-linux-gnu-gcc -c -Wsign-compare -DNDEBUG     -O3 -pipe -march=tigerlake -fwrapv -flto -fuse-linker-plugin -ffat-lto-objects -flto-partition=none -g -std=c11 -Wextra -Wno-unused-parameter -Wno-missing-field-initializers -Wstrict-prototypes -Werror=implicit-function-declaration -fvisibility=hidden  -I./Include/internal  -I. -I./Include -I/usr/include/ncursesw  -fPIC -DPy_BUILD_CORE -o Programs/python.o ./Programs/python.c`

**Comment by space-nuko** [comment]
Same error, my program seems to run the first few lines and then it gives up later, it ends up spawning subprocesses (`pip`) to install dependencies

```
Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
Commit hash: 8e439067bc6d197b4b672438a0ca6114d3cc7a6c
Installing sd-dynamic-prompts requirements.txt



Error: Failed to find a python interpreter in the .data section
```

- Windows Server 2022
- Python 3.10.9
- py-spy 0.3.14 (pip)

**Comment by notwopr** [comment]
I get the same error.  I installed python 3.10 directly from the python.org site.  i used pip install to install py-spy.

**Comment by notwopr** [comment]
> Same error, my program (https://github.com/AUTOMATIC1111/stable-diffusion-webui) seems to run the first few lines and then it gives up later, it ends up spawning subprocesses (`pip`) to install dependencies
> 
> ```
> # py-spy top -- %PYTHON% launch.py --skip-install
> Python 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
> Commit hash: 8e439067bc6d197b4b672438a0ca6114d3cc7a6c
> Installing sd-dynamic-prompts requirements.txt
> 
> 
> 
> Error: Failed to find a python interpreter in the .data section
> ```
> 
> * Windows Server 2022
> * Python 3.10.9
> * py-spy 0.3.14 (pip)
> 
> EDIT: Fixed it, issue was I didn't install my python with debugging symbols (`.pdb`s). When you launch the Python installer it gives you the option to download them under "Custom Install". Reading the docs, if py-spy doesn't detect the `.pdb` it tries to scan the binary for whatever it needs, fruitlessly

thank you for this.  i had the same .data error.  i just reinstalled python with the symbols and it works now. Thank you!

**Comment by jkriegshauser** [comment]
I too was having this same issue on Windows with a custom built python 3.10 binary running embedded in a very large application. Once I dropped the symbols alongside the DLL it worked like a charm!

As a suggestion, it might be good to log about missing symbols with this error

**Comment by benfred** [comment]
This should be fixed in the lastest release (0.4.0) - by https://github.com/benfred/py-spy/pull/543 and https://github.com/benfred/py-spy/pull/718. You shouldn't need symbols anymore to profile python 3.10+ on windows
